pkg/libhelm/binary/search_repo.go

```diff
@@ -64,6 +65,10 @@ func (hbpm *helmBinaryPackageManager) SearchRepo(searchRepoOpts options.SearchRe
 		}
 	}
 
+	if client.CheckRedirect != nil {
```

Comment:
This value is normally nil, it allows it to be overridden from EE if we wanted to add extra features over CE in the future

Reply:
Since the client is passed in with this blank, it'll use a default CheckRedirect included in the pkg.  This is a very common override pattern in go and allows more flexibility.

Reply:
You were right.  The logic should have been client.CheckRedirect == nil.   I must have added this after doing my testing.  It's definitely being called now.